src/common/framework/params.ts

```diff
@@ -1,112 +1,131 @@
-import {
-  ParamArgument,
-  ParamSpec,
-  ParamSpecIterable,
-  ParamSpecIterator,
-  paramsEquals,
-} from './params_utils.js';
+import { ParamSpec, ParamSpecIterable, paramsEquals } from './params_utils.js';
 import { assert } from './util/util.js';
 
-export function poptions(name: string, values: ParamArgument[]): POptions {
-  return new POptions(name, values);
-}
-export function pbool(name: string): POptions {
-  return new POptions(name, [false, true]);
-}
-export function pexclude(params: ParamSpecIterable, exclude: ParamSpecIterable): PExclude {
-  return new PExclude(params, exclude);
+export function poptions<Name extends string, V>(
+  name: Name,
+  values: Iterable<V>
+): string extends Name ? never : Iterable<{ [name in Name]: V }> {
```

Comment:
It was supposed to prevent calling poptions with `Name = string` (making sure it's a strict subtype of `string`; really we want to know it's a single string literal type). But I just tried it and I guess it's not serving its intended purpose...

Reply:
Actually it does seem to be working correctly. `poptions('x' as string, [])` is type `never`. This interacts slightly oddly with `.params()` though.

Reply:
And if you do poptions('x' as 'x' | 'y', []) it will think that both 'x' and 'y' are valid keys. whoops.

Reply:
Fixed with magic